tonic/src/transport/service/connector.rs

```diff
@@ -80,16 +80,26 @@ where
         #[cfg(feature = "tls-roots-common")]
         let tls = self.tls_or_default(uri.scheme_str(), uri.host());
 
+        let is_https = uri.scheme_str() == Some("https");
         let connect = self.inner.make_connection(uri);
 
         Box::pin(async move {
+            #[cfg(not(feature = "tls"))]
+            {
+                if is_https {
+                    return Err("connecting to HTTPS without TLS enabled".to_owned().into());
```

Comment:
Should we make this an actual error type?

Reply:
Seemed unlikely to me that code would be handling this, but I'm happy to change it if you prefer it the other way.

Reply:
Yeah lets make it a concrete error or error variant?

Reply:
Like this?